magic_bridge/ic/src/dip20_proxy/src/common/magic.rs

```diff
@@ -0,0 +1,38 @@
+use async_trait::async_trait;
+use ic_cdk::call;
+use ic_cdk::export::candid::Principal;
+
+use crate::common::types::TxError;
+
+#[async_trait]
+pub trait Magic {
+    async fn get_canister(&self, erc20_addr_pid: Principal) -> Result<Principal, TxError>;
+}
+
+#[async_trait]
+impl Magic for Principal {
+    async fn get_canister(&self, erc20_addr_pid: Principal) -> Result<Principal, TxError> {
+        let get_canister: (Result<Option<Principal>, String>,) =
+            match call(*self, "get_canister", (erc20_addr_pid,)).await {
+                Ok(res) => res,
+                Err((code, err)) => {
+                    return Err(TxError::Other(format!(
+                        "RejectionCode: {:?}\n{}",
+                        code, err
+                    )))
+                }
+            };
+
+        match get_canister {
+            (Ok(Some(canister_id)),) => Ok(canister_id),
+            (Ok(None),) => Err(TxError::Other(format!(
```

Comment:
we could perhaps combine the `error` and `none`

Reply:
I like the idea of having them separated so we could differentiate an error and a not existing canister

Reply:
But the method returns a result of either and ok or err no?

Reply:
fixed 👍